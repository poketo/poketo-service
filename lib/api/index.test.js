import poketo from '.';

describe('poketo', () => {
  describe('constructUrl', () => {
    it('returns a site-specific url', () => {
      expect(poketo.constructUrl('mangakakalot', 'urami_koi_koi_urami_koi')).toEqual('http://mangakakalot.com/manga/urami_koi_koi_urami_koi');
      expect(poketo.constructUrl('meraki-scans', 'senryu-girl', '5')).toEqual('http://merakiscans.com/senryu-girl/5');
    });
  });

  describe('getChapter', () => {
    it('returns a chapter', async () => {
      expect.assertions(2);

      await expect(poketo.getChapter('http://merakiscans.com/senryu-girl/2')).resolves.toMatchSnapshot();
      await expect(poketo.getChapter('http://helveticascans.com/r/read/talentless-nana/en/1/2/page/1')).resolves.toMatchSnapshot();
    });

    it('throws for invalid urls', async () => {
      expect.assertions(3);

      await expect(poketo.getChapter('banana')).rejects.toThrow(poketo.UnsupportedSiteError);
      await expect(poketo.getChapter('http://google.com')).rejects.toThrow(poketo.UnsupportedSiteError);
      await expect(poketo.getChapter('http://helveticascans.com/ba/nan/na')).rejects.toThrow(poketo.InvalidUrlError);
    });
  });
});
