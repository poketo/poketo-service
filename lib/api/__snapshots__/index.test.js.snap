// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`poketo getChapter returns a chapter 1`] = `
Object {
  "id": "meraki-scans:senryu-girl:2",
  "pages": Array [
    Object {
      "height": 700,
      "id": 0,
      "url": "http://merakiscans.com/wp-content/manga/21/2/00.jpg",
      "width": 600,
    },
    Object {
      "height": 1600,
      "id": 1,
      "url": "http://merakiscans.com/wp-content/manga/21/2/001.jpg",
      "width": 1114,
    },
    Object {
      "height": 1495,
      "id": 2,
      "url": "http://merakiscans.com/wp-content/manga/21/2/01.jpg",
      "width": 1000,
    },
    Object {
      "height": 1495,
      "id": 3,
      "url": "http://merakiscans.com/wp-content/manga/21/2/02.jpg",
      "width": 1000,
    },
    Object {
      "height": 1495,
      "id": 4,
      "url": "http://merakiscans.com/wp-content/manga/21/2/03.jpg",
      "width": 1000,
    },
    Object {
      "height": 1495,
      "id": 5,
      "url": "http://merakiscans.com/wp-content/manga/21/2/04.jpg",
      "width": 1000,
    },
    Object {
      "height": 1495,
      "id": 6,
      "url": "http://merakiscans.com/wp-content/manga/21/2/05.jpg",
      "width": 1000,
    },
    Object {
      "height": 1495,
      "id": 7,
      "url": "http://merakiscans.com/wp-content/manga/21/2/06.jpg",
      "width": 1000,
    },
    Object {
      "height": 1495,
      "id": 8,
      "url": "http://merakiscans.com/wp-content/manga/21/2/07.jpg",
      "width": 1000,
    },
    Object {
      "height": 1495,
      "id": 9,
      "url": "http://merakiscans.com/wp-content/manga/21/2/08.jpg",
      "width": 1000,
    },
    Object {
      "height": 1495,
      "id": 10,
      "url": "http://merakiscans.com/wp-content/manga/21/2/09.jpg",
      "width": 1000,
    },
    Object {
      "height": 1495,
      "id": 11,
      "url": "http://merakiscans.com/wp-content/manga/21/2/10.jpg",
      "width": 1000,
    },
    Object {
      "height": 1495,
      "id": 12,
      "url": "http://merakiscans.com/wp-content/manga/21/2/11.jpg",
      "width": 1000,
    },
    Object {
      "height": 1495,
      "id": 13,
      "url": "http://merakiscans.com/wp-content/manga/21/2/12.jpg",
      "width": 1000,
    },
  ],
  "url": "http://merakiscans.com/senryu-girl/2",
}
`;

exports[`poketo getChapter returns a chapter 2`] = `
Object {
  "id": "helvetica-scans:talentless-nana:en/1/2",
  "pages": Array [
    Object {
      "height": 900,
      "id": 3801,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0000.png",
      "width": 600,
    },
    Object {
      "height": 1280,
      "id": 3802,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0089 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3803,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0090 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3804,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0091 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3805,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0092 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3806,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0093 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3807,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0094 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3808,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0095 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3809,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0096 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3810,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0097 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3811,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0098 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3812,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0099 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3813,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0100 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3814,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0101 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3815,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0102 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3816,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0103 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3817,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0104 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3818,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0105 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3819,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0106 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3820,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0107 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3821,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0108 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3822,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0109 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3823,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0110 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3824,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0111 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3825,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0112 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3826,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0113 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3827,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0114 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3828,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0115 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3829,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0116 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3830,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0117 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3831,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0118 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3832,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0119 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3833,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0120 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3834,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0121 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3835,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0122 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3836,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0123 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3837,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0124 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3838,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0125 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3839,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0126 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3840,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0127 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3841,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0128 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3842,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0129 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3843,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0130 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3844,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0131 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3845,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0132 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3846,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0133 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3847,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0134 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3848,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0135 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3849,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0136 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3850,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0137 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3851,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0138 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3852,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0139 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3853,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0140 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3854,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0141 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3855,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0142 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3856,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0143 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3857,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0144 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3858,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0145 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3859,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0146 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3860,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0147 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3861,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0148 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3862,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0149 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3863,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0150 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3864,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0151 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3865,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0152 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3866,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0153 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3867,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0154 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3868,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0155 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3869,
      "url": "http://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0156 copy.png",
      "width": 900,
    },
  ],
  "url": "http://helveticascans.com/r/read/talentless-nana/en/1/2/page/1",
}
`;
